subprojects/build-cache/src/main/java/org/gradle/caching/internal/controller/NextGenBuildCacheController.java

```diff
@@ -216,6 +233,64 @@ public ImmutableSortedMap<String, FileSystemSnapshot> getResultingSnapshots() {
         });
     }
 
+    private FileSystemLocationSnapshot getSnapshot(File root, List<ManifestEntry> entries) {
+        ManifestEntry rootEntry = entries.get(0);
+        switch (rootEntry.getType()) {
+            case Directory:
+                return getDirectorySnapshot(root, entries);
+            case RegularFile:
+                File file = new File(root, rootEntry.getRelativePath());
+                return getFileSnapshot(rootEntry, file);
```

Comment:
❓ When the whole tree is a single file, then `root` should already point to the file, no?